src/hooks/useNetInfo.ts

```diff
@@ -0,0 +1,14 @@
+import NetInfo from '@react-native-community/netinfo';
+import {useEffect, useState} from 'react';
+
+export function useNetInfo() {
+  const [networkStatus, setNetworkStatus] = useState<boolean>(false);
+
+  useEffect(() => {
+    NetInfo.addEventListener((state) => {
+      setNetworkStatus(state.isConnected ? true : false);
+    });
+  }, [networkStatus]);
```

Comment:
```suggestion
  }, []);
```
We are attaching a listener here, No need to add the internal state as a dependency of the hook.